specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+        ajax<BackendStatsResult>('/statistics/collection/global/', {
+          method: 'GET',
+          headers: {
+            Accept: 'application/json',
+          },
+        }).then(({ data }) => data),
+      []
+    ),
+    false
+  );
+  return backendStatObject;
+}
+
+function useStatsSpec(): IR<
+  IR<{
+    readonly label: string;
+    readonly items: StatCategoryReturn;
+  }>
+> {
+  const backEndResult = useBackendApi();
+  return React.useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(statsSpec).map(([pageName, pageStatSpec]) => [
+          pageName,
+          Object.fromEntries(
+            Object.entries(pageStatSpec).map(
+              ([categoryName, { label, categories }]) => [
+                categoryName,
+                {
+                  label,
+                  items: (
+                    categories as (
+                      backendStatResult:
+                        | BackendStatsResult[typeof categoryName]
+                        | undefined
+                    ) => StatCategoryReturn
+                  )(backEndResult?.[categoryName]),
+                },
+              ]
+            )
+          ),
+        ])
+      ),
+    [backEndResult]
+  );
+}
+
+function useDefaultStatsToAdd(
+  layout: {
+    readonly label: string;
+    readonly categories: RA<{
+      readonly label: string;
+      readonly items: RA<CustomStat | DefaultStat>;
+    }>;
+  },
+  defaultLayout: StatLayout
+): StatLayout {
+  return React.useMemo((): StatLayout => {
+    const listToUse = layout.categories.flatMap(({ items }) =>
+      items.filter((item): item is DefaultStat => item.type === 'DefaultStat')
+    );
+    return defaultLayout.map((defaultLayoutPage) => ({
+      label: defaultLayoutPage.label,
+      categories: defaultLayoutPage.categories
+        .map(({ label, items }) => ({
+          label,
+          items: items.filter(
+            (defaultItem) =>
+              defaultItem.type === 'DefaultStat' &&
+              !listToUse.some(
+                ({ pageName, categoryName, itemName }) =>
+                  pageName === defaultItem.pageName &&
+                  categoryName === defaultItem.categoryName &&
+                  itemName === defaultItem.itemName
+              )
+          ),
+        }))
+        .filter(({ items }) => items.length > 0),
+    }));
+  }, [layout, defaultLayout]);
+}
+
+export function StatsPage(): JSX.Element {
+  const [customLayout, setLayout] = usePref(
+    'statistics',
+    'appearance',
+    'layout'
+  );
+  const statsSpec = useStatsSpec();
+
+  const defaultLayout = useDefaultLayout(statsSpec);
+  const layout = customLayout ?? defaultLayout;
+  const [state, setState] = React.useState<
+    | State<
+        'EditingState',
+        {
+          readonly addingItem:
+            | {
+                readonly pageIndex: number;
+                readonly categoryIndex: number;
+              }
+            | undefined;
+        }
```

Comment:
A reminder to implement a `Cancel` button that would cancel the changes made since you pressed the `Edit` button